lib/modem_info/modem_info_json.c

```diff
@@ -97,67 +45,181 @@ static int json_add_str(cJSON *parent, const char *str, const char *item)
 	return json_add_obj(parent, str, json_str);
 }
 
-int modem_info_json_string_get(char *buf)
+static int json_add_data(struct lte_param *param, cJSON *json_obj)
 {
-	u16_t param_value;
-	size_t len = 0;
-	size_t total_len = 0;
-	int ret = 0;
-	char data_buffer[MODEM_INFO_MAX_RESPONSE_SIZE] = {0};
 	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
-	cJSON *data_obj = cJSON_CreateObject();
 	enum at_param_type data_type;
+	int total_len = 0;
+	int ret;
+
+	memset(data_name, 0, MODEM_INFO_MAX_RESPONSE_SIZE);
+	ret = modem_info_name_get(param->type,
+				data_name);
+	if (ret < 0) {
+		LOG_DBG("Data name not obtained: %d", ret);
+		return -EINVAL;
+	}
+
+	data_type = modem_info_type_get(param->type);
+
+	if (data_type < 0) {
+		return -EINVAL;
+	}
+
+	if (data_type == AT_PARAM_TYPE_STRING &&
+	    param->type != MODEM_INFO_AREA_CODE) {
+		total_len += strlen(param->string);
+		ret += json_add_str(json_obj, data_name, param->string);
+	} else {
+		total_len += sizeof(u16_t);
+		ret += json_add_num(json_obj, data_name, param->value);
+	}
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	return total_len;
+}
+
+static int network_data_add(struct network_param *network, cJSON *json_obj)
+{
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
+	int total_len;
+	int ret;
+	int len;
+
+	static const char lte_string[]	 = "LTE-M";
+	static const char nbiot_string[] = "NB-IoT";
+	static const char gps_string[]	 = " GPS";
+
+	if (network == NULL || json_obj == NULL) {
+		return -EINVAL;
+	}
+
+	total_len = json_add_data(&network->current_band, json_obj);
+	total_len += json_add_data(&network->sup_band, json_obj);
+	total_len += json_add_data(&network->area_code, json_obj);
+	total_len += json_add_data(&network->current_operator, json_obj);
+	total_len += json_add_data(&network->ip_address, json_obj);
+	total_len += json_add_data(&network->ue_mode, json_obj);
+
+	len = modem_info_name_get(network->cellid_hex.type, data_name);
+	data_name[len] =  '\0';
+	ret = json_add_num(json_obj, data_name, network->cellid_dec);
+
+	if (ret) {
+		LOG_DBG("Unable to add the cell ID.");
+	} else {
+		total_len += sizeof(double);
+	}
+
+	if (network->lte_mode.value == 1) {
+		strcat(network->network_mode, lte_string);
+		total_len += sizeof(lte_string);
+	} else if (network->nbiot_mode.value == 1) {
+		strcat(network->network_mode, nbiot_string);
+		total_len += sizeof(nbiot_string);
+	}
+
+	if (network->gps_mode.value == 1) {
+		strcat(network->network_mode, gps_string);
+		total_len += sizeof(gps_string);
+	}
+
+	ret = json_add_str(json_obj, "ntwrkMode", network->network_mode);
+
```

Comment:
In general, if you set some variable and then verify it, we do not insert an empty line in between in Zephyr.

Reply:
Fixed for all instances I could find it.